packages/base/src/commands.ts

```diff
@@ -51,6 +51,62 @@ function loadKeybindings(commands: CommandRegistry, keybindings: any[]) {
   });
 }
 
+function downloadFile(content: BlobPart, fileName: string, mimeType: string) {
+  const blob = new Blob([content], { type: mimeType });
+  const url = URL.createObjectURL(blob);
+  const downloadLink = document.createElement('a');
+  downloadLink.href = url;
+  downloadLink.download = fileName;
+  document.body.appendChild(downloadLink);
+  downloadLink.click();
+  document.body.removeChild(downloadLink);
+}
+
+function getSelectedLayer(
+  tracker: JupyterGISTracker
+): { model: IJupyterGISModel; selectedLayer: any; sources: any } | null {
+  const model = tracker.currentWidget?.model as IJupyterGISModel;
+  if (!model) {
+    return null;
+  }
+
+  const localState = model.sharedModel.awareness.getLocalState();
+  if (!localState || !localState['selected']?.value) {
+    return null;
+  }
+
+  const selectedLayerId = Object.keys(localState['selected'].value)[0];
+  const layers = model.sharedModel.layers ?? {};
+  const sources = model.sharedModel.sources ?? {};
+
+  const selectedLayer = layers[selectedLayerId];
+  if (!selectedLayer || !selectedLayer.parameters) {
+    return null;
+  }
+
+  return { model, selectedLayer, sources };
+}
+
+async function getGeoJSONString(
+  source: any,
+  model: IJupyterGISModel
+): Promise<string | null> {
+  if (source.parameters.path) {
+    const fileContent = await loadFile({
+      filepath: source.parameters.path,
+      type: source.type,
```

Comment:
You could have a check on the `source.type`, depending on the source type we can know if the file can be loaded as GeoJSON

Reply:
Done in [c10c4a0](https://github.com/geojupyter/jupytergis/pull/528/commits/c10c4a0aca5f64003be5dd0b5fa4ba326b017505)